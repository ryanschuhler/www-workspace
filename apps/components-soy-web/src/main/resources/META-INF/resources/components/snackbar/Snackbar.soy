{namespace Snackbar}

/**
 * Toolbar
 */
{template .render}
  {@param alignStart: bool}

  {let $snackbarAttributes kind="attributes"}
    aria-live="assertive"
    aria-atomic="true"
    aria-hidden="true"
    class="mdc-snackbar

    {if $alignStart}
      {sp}mdc-snackbar--align-start
    {/if}
    "

    ref="mdcSnackbar"
  {/let}


  <div {$snackbarAttributes}>
    <div class="mdc-snackbar__text"></div>
    <div class="mdc-snackbar__action-wrapper">
      <button type="button" class="mdc-snackbar__action-button" aria-hidden="false"></button>
    </div>
  </div>
{/template}