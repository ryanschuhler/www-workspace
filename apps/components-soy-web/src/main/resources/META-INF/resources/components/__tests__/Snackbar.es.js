'use strict';

import Snackbar from '../snackbar/Snackbar.es';

let component;

describe('Snackbar', () => {
  afterEach(() => {
    if (component) {
      component.dispose();
    }
  });

  beforeAll(() => {
		window.Liferay = {
			mdc: {
				snackbar: {
					MDCSnackbar: {
            attachTo: jest.fn()
          }
				}
			},
			once: jest.fn()
    };
  });

  it('should render checked', () => {
    component = new Snackbar({
      actionText: 'barr',
      alignStart: true,
      message: 'foo'
    });

    expect(component).toMatchSnapshot();
  });
});