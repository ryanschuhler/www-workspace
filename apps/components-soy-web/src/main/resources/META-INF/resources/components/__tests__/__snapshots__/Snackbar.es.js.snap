// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar should render checked 1`] = `
<div aria-live="assertive" aria-atomic="true" aria-hidden="true" class="mdc-snackbar mdc-snackbar--align-start" ref="mdcSnackbar">
  <div class="mdc-snackbar__text"></div>
  <div class="mdc-snackbar__action-wrapper">
    <button type="button" class="mdc-snackbar__action-button" aria-hidden="false"></button>
  </div>
</div>
`;
